apps/web/src/pages/templates/editor_v2/TemplateStepEditorV2.tsx

```diff
@@ -37,13 +41,29 @@ export const WorkflowsStepEditorPageV2 = () => {
   } = useMutation<any, any, any>((data) => api.post('/v1/echo/preview/' + workflow?.name + '/' + stepId, data));
 
   const title = step?.stepId;
+  const stepType: ChannelTypeEnum | undefined = step?.template?.type;
 
-  const navigate = useNavigate();
-  const handleTestClick = () => {
-    // TODO: this is just a temporary step for connecting the prototype
-    navigate(ROUTES.STUDIO_FLOWS_TEST_STEP);
+  const handleTestClick = async () => {
+    try {
+      await testSendEmailEvent({
+        stepId,
+        workflowId: (workflow as any)?.rawData?.workflowId,
+        contentType: 'customHtml',
+        subject: '',
+        payload,
+        inputs,
+        to: currentUser?.email || '',
+        bridge: true,
+        content: '',
+      });
+      successMessage('Test sent successfully!');
```

Comment:
The copywriting in the `successMessage` needs to be more accurate as we didn't send a test.
```suggestion
      successMessage('Test run has started!');
```